cmd/agent/main.go

```diff
@@ -1,3 +1,171 @@
 package main
 
-func main() {}
+import (
+	"flag"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"reflect"
+	"runtime"
+	"time"
+
+	"github.com/caarlos0/env/v11"
+)
+
+type metrics struct {
+	Alloc         float64
+	BuckHashSys   float64
+	Frees         float64
+	GCCPUFraction float64
+	GCSys         float64
+	HeapAlloc     float64
+	HeapIdle      float64
+	HeapInuse     float64
+	HeapObjects   float64
+	HeapReleased  float64
+	HeapSys       float64
+	LastGC        float64
+	Lookups       float64
+	MCacheInuse   float64
+	MCacheSys     float64
+	MSpanInuse    float64
+	MSpanSys      float64
+	Mallocs       float64
+	NextGC        float64
+	NumForcedGC   float64
+	NumGC         float64
+	OtherSys      float64
+	PauseTotalNs  float64
+	StackInuse    float64
+	StackSys      float64
+	Sys           float64
+	TotalAlloc    float64
+	RandomValue   float64
+	PollCount     int64
+}
+
+const DA = "localhost:8080"
+const DPI = 2
+const DRI = 10
```

Comment:
Хорошая идея использовать константы для значений по-умолчанию, но я рекомендовал бы дать им более информативные имена:

```suggestion
const (
        AddressDefault = "localhost:8080"
        PollIntervalDefault = 2
        ReportIntervalDefault = 10
)
```